x-pack/plugins/fleet/common/types/models/agent.ts

```diff
@@ -421,3 +438,15 @@ export interface ActionStatusOptions {
   page?: number;
   perPage?: number;
 }
+
+export interface AgentUpgradeDetails {
+  target_version: string;
+  action_id: string;
+  state: AgentUpgradeStateType;
+  metadata: {
+    scheduled_at?: string;
+    download_percent?: number;
+    failed_state?: string; // TODO: create type with possible failed states
```

Comment:
@ycombinator 👋 Are these failed states known yet? As far as I understand they are not used by the UI at the moment, so if that's correct this might be enough for now.

Reply:
The type of `failed_state` should be `AgentUpgradeStateType`.  The idea is for this field to capture the state the upgrade was in right before it failed.  Hope that makes sense!

Reply:
Ah makes sense! Thanks 👍